interop/xds/client/Dockerfile

```diff
@@ -16,7 +16,7 @@
 # following command from grpc-go directory:
 # docker build -t <TAG> -f interop/xds/client/Dockerfile .
 
-FROM golang:1.21-alpine as build
```

Comment:
Go 1.23 is out now -- should we go straight to that version instead?

Reply:
Updated to `go1.23`. Also changed remaining Dockerfiles in the repo and `test/kokoro/xds.sh` to use `go1.23`.

Reply:
:+1: Then we won't need to update again in 6 months.